### ISSUE TITLE
Using interpolate to respond to transY position actually changes transY position

### ISSUE THREAD

**Issue by pribeh** [kmagiera/react-native-reanimated#184]
We've created a draggable / snappable bottom sheet (we call it a dashboard) based off of the snappable example. We're trying to animate the backgroundColor of the sheet as the user drags the sheet upward. We're using interpolate with a range to do this but it ends up shifting the actual position of the sheet instead of responding to it. Here's a snack which reproduces this:

https://snack.expo.io/@pribeh/transy-affected-by-interpolate

Switch out the dashboardBackgroundColor for '#fff' and you'll see the dashboard move back into the correct position.

Maybe we're just misunderstanding how to actually make the backgroundColor respond to the transY position of the sheet. Any help would be much appreciated as we're really stuck on this.

Here is the code for reference:

```
import React, { Component } from 'react';
import { StyleSheet, View, Dimensions, ScrollView, Text } from 'react-native';
import { PanGestureHandler, State } from 'react-native-gesture-handler';
import Animated from 'react-native-reanimated';

const screen = Dimensions.get('window');

// setInterval(() => {
//   let iters = 1e8,
//     sum = 0;
//   while (iters-- > 0) sum += iters;
// }, 300);

const {
  set,
  cond,
  eq,
  add,
  multiply,
  lessThan,
  spring,
  startClock,
  stopClock,
  clockRunning,
  sub,
  defined,
  Value,
  Clock,
  event,
  interpolate,
  color,
  Extrapolate
} = Animated;

function runSpring(clock, value, velocity, dest) {
  const state = {
    finished: new Value(0),
    velocity: new Value(0),
    position: new Value(0),
    time: new Value(0),
  };

  const config = {
    damping: 14,
    mass: 1,
    stiffness: 121.6,
    overshootClamping: false,
    restSpeedThreshold: 0.001,
    restDisplacementThreshold: 0.001,
    toValue: new Value(0),
  };

  return [
    cond(clockRunning(clock), 0, [
      set(state.finished, 0),
      set(state.velocity, velocity),
      set(state.position, value),
      set(config.toValue, dest),
      startClock(clock),
    ]),
    spring(clock, state, config),
    cond(state.finished, stopClock(clock)),
    state.position,
  ];
}

class Snappable extends Component {
  constructor(props) {
    super(props);

    const TOSS_SEC = 0.2;

    const dragY = new Value(0);
    const state = new Value(0);
    const dragVY = new Value(0);

    this._onGestureEvent = event([
      { nativeEvent: { translationY: dragY, velocityY: dragVY, state: state } },
    ]);

    const transY = new Value(0);
    const prevDragY = new Value(0);

    const clock = new Clock();

    // If transX has not yet been defined we stay in the center (value is 0).
    // When transX is defined, it means drag has already occured. In such a case
    // we want to snap to -100 if the final position of the block is below 0
    // and to 100 otherwise.
    // We also take into account gesture velocity at the moment of release. To
    // do that we calculate final position of the block as if it was moving for
    // TOSS_SEC seconds with a constant speed the block had when released (dragVX).
    // So the formula for the final position is:
    //   finalX = transX + TOSS_SEC * dragVelocityX
    //
    const snapPoint = cond(
      lessThan(add(transY, multiply(TOSS_SEC, dragVY)), 0),
      (screen.height / 2) - (screen.height / 3),
      screen.height / 2 + 150
    );

    this._transY = cond(
      eq(state, State.ACTIVE),
      [
        stopClock(clock),
        set(transY, add(transY, sub(dragY, prevDragY))),
        set(prevDragY, dragY),
        transY,
      ],
      [
        set(prevDragY, 0),
        set(
          transY,
          cond(defined(transY), runSpring(clock, transY, dragVY, snapPoint),
          screen.height / 2 + 150)
        ),
      ]
    );
  }
  render() {
    const { children, ...rest } = this.props;
    return (
      <PanGestureHandler
        {...rest}
        activeOffsetY={[-20, 20]}
        maxPointers={1}
        onGestureEvent={this._onGestureEvent}
        onHandlerStateChange={this._onGestureEvent}>
        <Animated.View style={{ transform: [{ translateY: this._transY }] }}>
          {children}
        </Animated.View>
      </PanGestureHandler>
    );
  }
}

export default class Example extends Component {
  static navigationOptions = {
    title: 'Snappable Example',
  };
  render() {
            const alpha = interpolate(this._transY, {
            inputRange: [screen.height / 2 - 100, screen.height -200, screen.height - 150, screen.height - 100],
            outputRange: [1, 1, 0.9, 0.8],
            extrapolate: Extrapolate.CLAMP
        });
        
        const dashboardBackgroundColor = color(255, 255, 255, alpha);
        
    return (
      <View style={styles.container}>
        <Snappable>
          <View style={[styles.box, {backgroundColor:'dashboardBackgroundColor'}]} />
        </Snappable>
      </View>
    );
  }
}

const BOX_SIZE = 100;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#000',
  },
  box: {
    width: screen.width,
    height: screen.height / 2 + 100,
    borderTopLeftRadius:10,
    borderTopRightRadius:10,
    borderColor: '#F5f5f5',
    alignSelf: 'center',
    backgroundColor: 'red',
  },
});
```

**Comment by pribeh** [comment]
okay, I resolved the main issue by moving the consts into constructor and making them into this.alpha and this.dashboardBackgroundColor. However, now the dashboard still responds to certain touchable events inside the pangesturehandler, causing the sheet to close or bounce. 

I noticed this was happening with other touchables completely outside of the component in question. Once I moved the interpolate functions inside constructor that stopped. Now, it only happens when using touchables inside the pangesturehandler.

Perhaps this is a gesturehandler issue, but none of this seemed to happen when using the animated library. I'm going to see if removing interpolate altogether removes the bounce with the touchables.
